File: bankingOnGolang/internal/model/Bank.go
package model

import ()

// ==============================================================
// Bank Declaration
// ==============================================================
type Bank struct {
	BaseModel
	Name                string
	LegalName           string
	SwiftBic            BIC
	HeadquartersCountry string
	Website             string
	Branches            []Branch             `gorm:"foreignKey:BranchesFromBankId"`
	Products            []BankingProduct     `gorm:"foreignKey:ProductsFromBankId"`
	Customers           []Customer           `gorm:"foreignKey:CustomersFromBankId"`
	Accounts            []Account            `gorm:"foreignKey:AccountsFromBankId"`
	PaymentCards        []PaymentCard        `gorm:"foreignKey:PaymentCardsFromBankId"`
	LoanAccounts        []LoanAccount        `gorm:"foreignKey:LoanAccountsFromBankId"`
	ExchangeRates       []ExchangeRate       `gorm:"foreignKey:ExchangeRatesFromBankId"`
	Consents            []Consent            `gorm:"foreignKey:ConsentsFromBankId"`
	ThirdPartyProviders []ThirdPartyProvider `gorm:"foreignKey:ThirdPartyProvidersFromBankId"`

	// parent associations as their child

}

File: src/app/models/Bank.ts


// Define collection and schema for Bank
export  Bank {
    name: string
    legalName: string
    swiftBic: BIC
    headquartersCountry: string
    website: string
    Branches:  [{ type: Schema.Types.ObjectId, ref: 'Branch' }]
    Products:  [{ type: Schema.Types.ObjectId, ref: 'BankingProduct' }]
    Customers:  [{ type: Schema.Types.ObjectId, ref: 'Customer' }]
    Accounts:  [{ type: Schema.Types.ObjectId, ref: 'Account' }]
    PaymentCards:  [{ type: Schema.Types.ObjectId, ref: 'PaymentCard' }]
    LoanAccounts:  [{ type: Schema.Types.ObjectId, ref: 'LoanAccount' }]
    ExchangeRates:  [{ type: Schema.Types.ObjectId, ref: 'ExchangeRate' }]
    Consents:  [{ type: Schema.Types.ObjectId, ref: 'Consent' }]
    ThirdPartyProviders:  [{ type: Schema.Types.ObjectId, ref: 'ThirdPartyProvider' }]
#
    collection: 'banks'
}
